### DISCUSSION TITLE
Retrieve non-HTTP (MLLP/HL7) requests in Symfony 5.4

### DISCUSSION THREAD

**Discussion by kilianpelissier** [symfony/symfony#50646] [Q&A] [answered]
Hello,

I am currently trying to retrieve MLLP requests on Symfony in order to retrieve and process messages in HL7. However, I can't find a way to retrieve the request as it is not in HTTP format.

I can see the request in the PHP console:

```
middleware_mllp | [Tue Jun 13 09:33:00 2023] 172.17.10.30:64618 Accepted
middleware_mllp | [Tue Jun 13 09:33:00 2023] 172.17.10.30:64618 Invalid request (Malformed HTTP request)
middleware_mllp | [Tue Jun 13 09:33:00 2023] 172.17.10.30:64618 Closing

```
I did a RequestListener and an ExceptionListener but obviously I'm not getting anything back

If anyone has an idea or has made something similar, I'm all ears.

Thanks in advance

**Comment by ghost** [comment]
WOW! That is very interesting. Non-http requests with Symfony. 
Can I ask you how do you run PHP? Symfony server, php -S, fpm? Second, does symfony app even fires or not and this request is invalid at the PHP server level because it does not follow http spec?

**Reply by kilianpelissier** [reply]
Yes, it's very interesting 😁
I run PHP with a PHP 8.1.19  Server, in fact i use a docker container with the bitnami/symfony image.
When the request is received nothing happens in symfony, the listener function is not called
The request being invalid but can it pass the level of PHP server?

Thanks

**Reply by ghost** [reply]
I dont think you will be able to use Symfony request and response objects and flow because it's all http based (even the component that holds these classes is called httpfoundation). I'm not sure how it works on PHP level but I think it doesn't because PHP server is not able to create global variables like $_GET etc. from the request . Mllp is a different protocol so I think you have to create custom sockets etc. I'm sure there are libraries for that.

**Reply by ghost** [reply]
Found a Mllp server implementation in PHP https://github.com/Pharma-Intelligence/MLLP but last commit is from 2015.

**Reply by ghost** [reply, marked as answer]
This is also interesting https://github.com/rivethealth/mllp-http. A proxy server that translates Mllp messages to http messages and vice versa

**Reply by kilianpelissier** [reply]
Thank you four your help, i will look at this in detail and i will probably install an MLLP to HTTP proxy server before my php server
